API: drop None rather than unset in JSON response

tombh opened the pull request:
This addresses #217 and #231.

I am still new to the project, so I may not have the full story. But it
started with a somewhat far-reaching merge to remove defaults from all
serialisable fields in the API: #198. This had the knock-on effect of
regressing progress notifications, as reported by @MatejKastak in #217.
One of their suggestions is to not use `exclude_unset=True`. Some months
later @dimbleby suggested that `exclude_none=True` would also fix the
issue[1]. Considering those 2 suggestions, let's replace
`exclude_unset=True` with `exclude_none=True`.

1. https://github.com/openlawlibrary/pygls/pull/198#issuecomment-1086864429

## Code review checklist (for code reviewer to complete)

- [x] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR)
- [x] Title summarizes what is changing
- [x] Commit messages are meaningful (see [this][commit messages] for details)
- [x] Tests have been included and/or updated, as appropriate
- [ ] Docstrings have been included and/or updated, as appropriate
- [ ] Standalone docs have been updated accordingly
- [x] [CONTRIBUTORS.md][contributors] was updated, as appropriate
- [ ] Changelog has been updated, as needed (see [CHANGELOG.md][changelog])

[changelog]: https://github.com/openlawlibrary/pygls/blob/master/CHANGELOG.md
[commit messages]: https://chris.beams.io/posts/git-commit/
[contributors]: https://github.com/openlawlibrary/pygls/blob/master/CONTRIBUTORS.md

tombh commented:
@MatejKastak and @dimbleby: would you like to comment on this fix?

dimbleby commented:
Looks familiar! sure, seems ok to me

tombh commented:
@dimbleby haha, yes _you're_ fix 😏 Thank you

@MatejKastak Thank you too, and thanks for waiting

Just waiting for the review, then LGTM 🚀

renatav commented on tests/lsp/test_progress.py:
```diff
@@ -0,0 +1,114 @@
+            CODE_LENS,
+            CodeLensOptions(
+                resolve_provider=False,
+                work_done_progress='token'
```
Any reason why `PROGRESS_TOKEN` is not used here?

renatav reviewed (commented):
LGTM, just one question

perrinjerome commented:
Can you also add something like:

Fixes https://github.com/openlawlibrary/pygls/issues/231

in the description so that this issue gets automatically fixed  ?

perrinjerome commented on CONTRIBUTORS.md:
```diff
@@ -15,3 +15,4 @@
 - [yorodm](https://github.com/yorodm)
 - [bollwyvl](https://github.com/bollwyvl)
 - [Dillan Mills](https://github.com/DillanCMills)
+- [Tom BH](https://github.com/tombh)
```
It seems it was not always followed earlier either, but this list is supposed to be sorted in alphabetical order ( https://github.com/openlawlibrary/pygls/blob/f53d6615e4561184b393d4b2c6b5629f31f33dbb/CONTRIBUTORS.md#contributors-alphabetical )

perrinjerome reviewed (approved):
Good to see this :)

perrinjerome commented:
> Can you also add something like:
> 
> Fixes #231
> 
> in the description so that this issue gets automatically fixed ?

Oh sorry, I did not realize the issue was already mentioned in the pull request description 😖.

That said, you can still change to using a keyword like **Fixes** if you wish, because it is understood by github ( https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword ) and

renatav commented:
@tombh I don't think that this issue is specific to this PR, but Python 3.6 Windows build is getting stuck. Would be good to know what is causing this even though we are planning to drop Python 3.6 support

tombh commented:
@perrinjerome I added the "Fixes" to the commit message and changed the contributors order. Thanks

@renatav Yes I saw that, and have had a play with it, it's actually intermittent and I've even seen it affect almost all items in the matrix, any version or OS, but yes more often Windows/Python 3.6. There's evidence of it happening before this PR, so as you say I don't think it's being introduced with this code. There's no lock file for the dependencies, so the only thing I can think is that a minor change in a dependent package has caused this. Which leads me to think that our next step should be #229, #223 and of course #235. So I vote for merging this PR as all the tests are passing, just rarely without some failure from the flakey bug.

renatav commented:
@tombh We usually use fixes in the PR comment, not sure that commit message with fixes will automatically close issues. I'll approve the PR
